### PR TITLE
PLAT-83172: Space added for Heading with line, cleanup of divider cruft

### PR DESCRIPTION
**Pull request by Djspaceg**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Heading text is too close to the line. It gives people anxiety; like driving really close to the cement barrier on the highway.

### Resolution
Added some space (according to the spec) between the text and the line, when the line is present. The design doc indicates that the space should be `12px`, however, the 12px spacing is based on the bottom of Korean characters, which don't include descenders or a taller-than-what's-visible line-height. Therefore this PR sets a `9px` padding, which accounts for a roughly `3px` inherent gap added by the line-height rule respecting taller glyphs and descenders. The net visual effect should be 12px total.

### Additional Considerations
Finally doing house-keeping on the ancient divider mixin, which should no longer be necessary with the built-in customizability of the newer `Heading` component.

### CODE REVIEW

File: packages/moonstone/Heading/Heading.module.less [hunk 1]
```diff
@@ -59,6 +59,7 @@
 
 		&.showLine {
 			border-bottom-color: @moon-heading-border-color;
+			padding-bottom: 9px;
```
Comment (teckliew) [inline packages/moonstone/Heading/Heading.module.less]: Should this be set outside of `applySkins` with its own variable?

Comment (Djspaceg) [reply packages/moonstone/Heading/Heading.module.less]: I suppose it could be, sure. I'll make that change real quick.

Comment (Djspaceg) [reply packages/moonstone/Heading/Heading.module.less]: Done.

**Comment by ryanjduffy** [comment]
There are a few other `@moon-divider-` references that remain. I'm assuming they should be removed as well but perhaps they were left intetionally.

image

**Comment by Djspaceg** [comment]
They were not left intentionally. I'll remove them and add them to the removed list after the merge conflict is resolved.

**Comment by Djspaceg** [comment]
Branch updated. Merge conflict resolved, divider variables removed and documented.

**Pull request merged**